src/index.tsx

```diff
@@ -1,52 +1,100 @@
-import React from 'react';
-import { IconKeys, icons } from './icons';
+import React, { CSSProperties } from 'react';
+import { BaseIconKeys, baseIcons as origBaseIcons } from './baseIcons';
+import {
+  IllustrativeIconKeys,
+  illustrativeIcons as origIllustrativeIcons
+} from './illustrativeIcons';
 import {
-  StaticIconKeys,
-  staticIcons,
   DoctypeIconKeys,
-  doctypeIcons
-} from './staticIcons';
-export { IconKeys } from './icons';
-export { StaticIconKeys, DoctypeIconKeys } from './staticIcons';
+  doctypeIcons as origDoctypeIcons
+} from './doctypeIcons';
+import {
+  ComponentIconKeys,
+  componentIcons as origComponentIcons
+} from './componentIcons';
+
+export { BaseIconKeys } from './baseIcons';
+export { IllustrativeIconKeys } from './illustrativeIcons';
+export { DoctypeIconKeys } from './doctypeIcons';
+export { ComponentIconKeys } from './componentIcons';
+
+export const baseIcons = Object.keys(origBaseIcons);
+export const illustrativeIcons = Object.keys(origIllustrativeIcons);
+export const doctypeIcons = Object.keys(origDoctypeIcons);
+export const componentIcons = Object.keys(origComponentIcons);
 
 const fallbackIcon = 'login';
+const fallbackStaticIcon = 'authorisation';
+const fallbackComponentIcon = 'toggle';
 
 function objValue<T, K extends keyof T>(obj: T, key: K) {
   return obj[key];
 }
 
-const getIcon = (icon: IconKeys | StaticIconKeys | DoctypeIconKeys) => {
-  const suomifiIcon =
-    icon in icons
-      ? objValue(icons, icon as IconKeys)
-      : icon in staticIcons
-      ? objValue(staticIcons, icon as StaticIconKeys)
-      : objValue(doctypeIcons, icon as DoctypeIconKeys);
-  return !!suomifiIcon ? suomifiIcon : objValue(icons, fallbackIcon);
-};
-
-export const allIcons = Object.keys(icons);
-export const allStaticIcons = Object.keys(staticIcons);
-export const allDoctypeIcons = Object.keys(doctypeIcons);
-
-export interface SuomifiIconInterface {
-  icon: IconKeys | StaticIconKeys | DoctypeIconKeys;
-  color?: string;
-  fill?: string;
+function fillProp<T extends FillPropsInterface>(props: T): CSSProperties {
+  const { color, fill: origFill } = props;
+  const fill = !!origFill ? origFill : color;
+  return !!fill ? { fill: fill } : {};
+}
+
+interface SuomifiIconBaseInterface<T> {
+  /** Icon name */
+  icon: T;
+  /** Custom classname to extend or customize */
   className?: string;
   // Allow passing unsupported custom props to SVG without providing an API
   [key: string]: any;
 }
 
+interface FillPropsInterface {
```

Comment:
Not really a problem, but maybe a useful convention that the interfaces should be introduced before they are used in the file.

Reply:
Agreed, fixed this. We should have linter rules in place to take care of this. Maybe even use the same linter settings for all repos.